cpl/module_block_data.F90

```diff
@@ -95,11 +96,12 @@ subroutine block_copy_1d_i4_to_2d_r8(destin_ptr, source_ptr, block, block_index,
       if (present(scale_factor)) factor = scale_factor
 !$omp parallel do private(ix,ib,jb,i,j)
       do ix = 1, block%blksz(block_index)
+        im = offset + ix - 1
```

Comment:
Should this im have a private omp attribute as other local indexes.

Reply:
Also the meaning of this index (im, a local variable) is different than im used in, for example GFS_typedefs where it means total number of columns. I wonder if this will create some confusion.

Reply:
Yes, I am adding this to all routines in that file now

Reply:
Good point ... I don't care what the name is and can rename it to whatever you want.

One thing to note is that a lot of the complicated logic here will go away. In the higher-level routine that calls these per-block copies, we can remove the loop over blocks and directly assign the arrays. I didn't want to make this change because I wanted b4b identical results and thus tried to avoid any unnecessary changes.

Reply:
Maybe ic, for column?

 But now I see in all io routines im is used, for example:
```
        im = Model%chunk_begin(nb)+ix-1

        clm_lake%T_snow(i,j) = Sfcprop%T_snow(im)
```
This im in io routines is the same as im in this block copy routines, correct? Is `offset` here equal to `Model%chunk_begin(nb)`?

Reply:
yes, correct - offset = Model%chunk_begin(nb)